atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/schema/stream/StreamTest.java

```diff
@@ -150,47 +180,206 @@ private long storeAndCheckByteStreams(int size) throws IOException {
+        Random rand = new Random();
+        rand.nextBytes(bytesToStore);
+
+        final long id = storeStream(bytesToStore, PtBytes.toBytes(megabytes));
+
+        Stopwatch timer = Stopwatch.createStarted();
+        InputStream stream = txManager.runTaskThrowOnConflict(t -> store.loadStream(t, id));
+        byte[] sample = new byte[1];
+        //noinspection ResultOfMethodCallIgnored
+        stream.read(sample);
+        timer.stop();
+        Assert.assertEquals(bytesToStore[0], sample[0]);
+        long millis = timer.elapsed(TimeUnit.MILLISECONDS);
+        System.out.println(String.format("Getting first byte of %dMB took %d milliseconds.", megabytes, millis));
+        return millis;
+    }
+
+    private long storeStream(byte[] bytesToStore, byte[] reference) {
+        final long id = timestampService.getFreshTimestamp();
+        txManager.runTaskWithRetry(t -> {
+            store.storeStreams(t, ImmutableMap.of(id, new ByteArrayInputStream(bytesToStore)));
+            store.markStreamAsUsed(t, id, reference);
+            return null;
+        });
+
+        return id;
+    }
+
+    private void verifyLoadingStreams(long id, byte[] bytesToStore) throws IOException {
+        verifyLoadStream(id, bytesToStore);
+        verifyLoadStreams(id, bytesToStore);
+        verifyLoadStreamAsFile(id, bytesToStore);
+    }
+
+    private void verifyLoadStreamAsFile(long id, byte[] bytesToStore) throws IOException {
         File file = txManager.runTaskThrowOnConflict(t -> store.loadStreamAsFile(t, id));
         Assert.assertArrayEquals(bytesToStore, FileUtils.readFileToByteArray(file));
     }
 
-    private void verifyLoadStreams(long id, byte[] bytesToStore, GenericStreamStore<Long> store) throws IOException {
+    private void verifyLoadStreams(long id, byte[] bytesToStore) throws IOException {
         Map<Long, InputStream> streams = txManager.runTaskThrowOnConflict(t ->
                 store.loadStreams(t, ImmutableSet.of(id)));
         assertStreamHasBytes(streams.get(id), bytesToStore);
     }
 
-    private void verifyLoadStream(long id, byte[] bytesToStore, GenericStreamStore<Long> store) throws IOException {
+    private void verifyLoadStream(long id, byte[] bytesToStore) throws IOException {
         InputStream stream = txManager.runTaskThrowOnConflict(t -> store.loadStream(t, id));
         assertStreamHasBytes(stream, bytesToStore);
     }
 
     private void assertStreamHasBytes(InputStream stream, byte[] bytes) throws IOException {
         byte[] streamAsBytes = IOUtils.toByteArray(stream);
         Assert.assertArrayEquals(bytes, streamAsBytes);
+        stream.close();
+    }
+
+    @Test
+    public void testOverwrite() throws IOException {
+        Random rand = new Random();
+        StreamTestTableFactory tableFactory = StreamTestTableFactory.of();
+
+        final byte[] reference = PtBytes.toBytes("ref");
+        final byte[] bytes1 = new byte[2 * StreamTestStreamStore.BLOCK_SIZE_IN_BYTES];
```

Comment:
we need some fencepost tests around the `BLOCK_SIZE_IN_BYTES` and `IN_MEMORY_THRESHOLD`